### PR TITLE
Add toString, equals, hashcode methods to entities

### LINKED ISSUE #510: Entity references in logs
some entities still appear as obj refs (e.g. `io.cloudslang.lang.entities.ResultNavigation@5b7d`) instead of toString representation

### PR DESCRIPTION
**Pull request by Bonczidai**
fixes #510 

they will also be logged properly

Signed-off-by: Bonczidai Levente <levente.bonczidai@hpe.com>

### CODE REVIEW

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/AsyncLoopStatement.java [hunk 1]
```diff
@@ -43,27 +41,4 @@ public String getVarName() {
         return varName;
     }
 
-    @Override
```
Comment (oritstone) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/AsyncLoopStatement.java]: Why did you move the toString(), equals() & hashcode() methods to the parent class? 
You miss the extending class's specific fields, such as `varName` in this case

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/LoopStatement.java [hunk 2]
```diff
@@ -37,4 +40,20 @@ protected LoopStatement() {
     public String getExpression() {
         return expression;
     }
+
```
Comment (oritstone) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/LoopStatement.java]: Why do we use these libraries? they use reflection to determine the fields of each object. It is probably slower than doing it manually.

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/AsyncLoopStatement.java (hunk 1 above)
Comment (Bonczidai) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/AsyncLoopStatement.java]: subclass fields are included - we only the method in superclass

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/LoopStatement.java (hunk 2 above)
Comment (Bonczidai) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/LoopStatement.java]: I prefer these methods over doing it manually, since
- we don't need to define each method in each subclass
- reduced code size
- standard output format
- can be considered best practice, better control over data

I like this approach, but we can go with explicit methods for each subclass - WDYT?

Comment (oritstone) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/LoopStatement.java]: I think that the elegance and code size is not enough in this case...
We will loose in the performance area for each time we need to compare 2 objects or toString() them, and I think it will probably happen a lot...

**Comment by Bonczidai** [comment]
@oritstone, how about this style? ->

```java
    @Override
    public String toString() {
        return "CompilationArtifact{" +
                "executionPlan=" + executionPlan +
                ", dependencies=" + dependencies +
                ", inputs=" + inputs +
                ", systemProperties=" + systemProperties +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass()) return false;

        CompilationArtifact that = (CompilationArtifact) o;

        return new EqualsBuilder()
                .append(executionPlan, that.executionPlan)
                .append(dependencies, that.dependencies)
                .append(inputs, that.inputs)
                .append(systemProperties, that.systemProperties)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37)
                .append(executionPlan)
                .append(dependencies)
                .append(inputs)
                .append(systemProperties)
                .toHashCode();
    }
```

**Comment by oritstone** [comment]
Seems OK. We should use standard equals() hashCode() & toString() methods for each class

**Comment by orius123** [comment]
Isn't there any `toString()` builder? For consistency.

**Comment by Bonczidai** [comment]
:+1:  there is, I used it in the commit, just didn't update the comment

**Comment by orius123** [comment]
Great minds..
